src/lib/transaction_logic/zkapp_command_logic.ml

```diff
@@ -1640,6 +1660,10 @@ module Make (Inputs : Inputs_intf) = struct
       Local_state.add_check local_state Global_excess_overflow
         Bool.(not global_excess_update_failed)
     in
+    (* store supply increase in global state *)
+    let global_state =
+      Global_state.set_supply_increase global_state local_state.supply_increase
```

Comment:
We should set the global state only if `is_last_account_update` and `local_state.success` is true, right? In failure cases, new accounts don't  get created

Reply:
Made that change.